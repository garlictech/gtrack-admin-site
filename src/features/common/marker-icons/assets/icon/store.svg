<?xml version="1.0" encoding="UTF-8"?>
<svg width="16px" height="16px" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 53.2 (72643) - https://sketchapp.com -->
    <title>icon/normal/store</title>
    <desc>Created with Sketch.</desc>
    <g id="icon/normal/store" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <path d="M11,6.97665103 C10.7058259,7.14682302 10.3642861,7.24421875 10,7.24421875 C9.63571388,7.24421875 9.29417405,7.14682302 9,6.97665103 L9,7.99360578e-15 L11,1.59872116e-14 L11,6.97665103 Z M13,6.97665103 L13,7.99360578e-15 L14.4685313,1.59872116e-14 L16,5.24421875 C16,6.34878125 15.1045625,7.24421875 14,7.24421875 C13.6357139,7.24421875 13.2941741,7.14682302 13,6.97665103 Z M7,6.97665103 C6.70582595,7.14682302 6.36428612,7.24421875 6,7.24421875 C5.63571388,7.24421875 5.29417405,7.14682302 5,6.97665103 L5,7.99360578e-15 L7,1.59872116e-14 L7,6.97665103 Z M3,6.97665103 C2.70582595,7.14682302 2.36428612,7.24421875 2,7.24421875 C0.8954375,7.24421875 0,6.34878125 0,5.24421875 L1.53146875,1.59872116e-14 L3,1.59872116e-14 L3,6.97665103 Z M14,8.28765625 C14.5189687,8.28765625 14.5759884,8.15696875 15.0039571,7.927 L15.0039571,16 L6,16 L6,9.99824524 L3,9.99824524 L3,16 L0.990717777,16 L0.990717777,7.927 C1.41871778,8.15696875 1.48103125,8.28765625 2,8.28765625 C2.76484375,8.28765625 3.46475,8.004 4,7.53646875 C4.53525,8.00396875 5.23515625,8.28765625 6,8.28765625 C6.76484375,8.28765625 7.46475,8.004 8,7.53646875 C8.53525,8.00396875 9.23515625,8.28765625 10,8.28765625 C10.7648437,8.28765625 11.46475,8.004 12,7.53646875 C12.53525,8.00396875 13.2351562,8.28765625 14,8.28765625 Z M8,13.0044765 L13.0068359,13.0044765 L13.0068359,9.99880386 L8,9.99880386 L8,13.0044765 Z" id="icon" fill="#7C7CF0" fill-rule="nonzero"></path>
    </g>
</svg>