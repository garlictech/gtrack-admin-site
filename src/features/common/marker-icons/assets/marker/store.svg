<?xml version="1.0" encoding="UTF-8"?>
<svg width="44px" height="56px" viewBox="0 0 44 56" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 53.2 (72643) - https://sketchapp.com -->
    <title>normal/store</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <filter x="-37.5%" y="-27.3%" width="175.0%" height="154.5%" filterUnits="objectBoundingBox" id="filter-1">
            <feOffset dx="0" dy="0" in="SourceAlpha" result="shadowOffsetOuter1"></feOffset>
            <feGaussianBlur stdDeviation="3" in="shadowOffsetOuter1" result="shadowBlurOuter1"></feGaussianBlur>
            <feColorMatrix values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.4 0" type="matrix" in="shadowBlurOuter1" result="shadowMatrixOuter1"></feColorMatrix>
            <feMerge>
                <feMergeNode in="shadowMatrixOuter1"></feMergeNode>
                <feMergeNode in="SourceGraphic"></feMergeNode>
            </feMerge>
        </filter>
    </defs>
    <g id="Icons" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="new-icons-normal" transform="translate(-219.000000, -240.000000)" fill-rule="nonzero">
            <g id="normal/store" filter="url(#filter-1)" transform="translate(225.000000, 246.000000)">
                <g id="pin/normal">
                    <path d="M16,-3.55271368e-15 C7.17755235,-3.55271368e-15 0,7.14846184 0,15.9350663 C0,26.8395171 14.3184326,42.8478882 14.9280525,43.5240452 C15.5006552,44.1592101 16.5003802,44.1580929 17.0719475,43.5240452 C17.6815674,42.8478882 32,26.8395171 32,15.9350663 C31.9998274,7.14846184 24.8223614,-3.55271368e-15 16,-3.55271368e-15 Z" id="Path" fill="#7C7CF0"></path>
                    <circle id="Oval" fill="#FFFFFF" cx="16" cy="16" r="13"></circle>
                </g>
                <g id="icon/normal/store" transform="translate(8.000000, 8.000000)" fill="#7C7CF0">
                    <path d="M11,6.97665103 C10.7058259,7.14682302 10.3642861,7.24421875 10,7.24421875 C9.63571388,7.24421875 9.29417405,7.14682302 9,6.97665103 L9,7.99360578e-15 L11,1.59872116e-14 L11,6.97665103 Z M13,6.97665103 L13,7.99360578e-15 L14.4685313,1.59872116e-14 L16,5.24421875 C16,6.34878125 15.1045625,7.24421875 14,7.24421875 C13.6357139,7.24421875 13.2941741,7.14682302 13,6.97665103 Z M7,6.97665103 C6.70582595,7.14682302 6.36428612,7.24421875 6,7.24421875 C5.63571388,7.24421875 5.29417405,7.14682302 5,6.97665103 L5,7.99360578e-15 L7,1.59872116e-14 L7,6.97665103 Z M3,6.97665103 C2.70582595,7.14682302 2.36428612,7.24421875 2,7.24421875 C0.8954375,7.24421875 0,6.34878125 0,5.24421875 L1.53146875,1.59872116e-14 L3,1.59872116e-14 L3,6.97665103 Z M14,8.28765625 C14.5189687,8.28765625 14.5759884,8.15696875 15.0039571,7.927 L15.0039571,16 L6,16 L6,9.99824524 L3,9.99824524 L3,16 L0.990717777,16 L0.990717777,7.927 C1.41871778,8.15696875 1.48103125,8.28765625 2,8.28765625 C2.76484375,8.28765625 3.46475,8.004 4,7.53646875 C4.53525,8.00396875 5.23515625,8.28765625 6,8.28765625 C6.76484375,8.28765625 7.46475,8.004 8,7.53646875 C8.53525,8.00396875 9.23515625,8.28765625 10,8.28765625 C10.7648437,8.28765625 11.46475,8.004 12,7.53646875 C12.53525,8.00396875 13.2351562,8.28765625 14,8.28765625 Z M8,13.0044765 L13.0068359,13.0044765 L13.0068359,9.99880386 L8,9.99880386 L8,13.0044765 Z" id="icon"></path>
                </g>
            </g>
        </g>
    </g>
</svg>