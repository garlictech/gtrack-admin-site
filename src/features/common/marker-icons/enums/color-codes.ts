// color names: https://www.colorhexa.com/
export enum EColorCodes {
  BLUE = '#3975D7',
  BROWN = '#9D7051',
  CYAN = '#5EC8BD',
  DARK_GREEN = '#118E4D',
  DARK_MAGENTA = '#74286B',
  GREY = '#A8A8A8',
  LIGHT_GREEN = '#66C547',
  ORANGE = '#FF8A22',
  PURPLE = '#8C4EB8',
  RED = '#F34648',
  // Custom
  HIGHLIGHTED = '#FF2905',
  // For admin
  PIN = '#2299E5'
}
