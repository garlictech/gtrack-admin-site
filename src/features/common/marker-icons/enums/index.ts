export * from './color-codes';
export * from './default-color-map';
export * from './icon-source';
export * from './icon-style';
