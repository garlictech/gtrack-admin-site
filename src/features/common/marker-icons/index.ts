export * from './marker-icons.module';
export * from './enums';
export * from './services';
