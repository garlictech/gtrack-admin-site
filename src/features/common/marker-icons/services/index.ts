import { Injectable } from '@angular/core';
import { SvgIconRegistryService } from 'angular-svg-icon';

@Injectable()
export class MarkerIconService {
  constructor(private readonly _iconReg: SvgIconRegistryService) {}

  getInlineSvg(path: string): any {
    console.log('PATH', path);
    this._iconReg.loadSvg(path).subscribe((element: SVGElement) => {
      console.log('element', element);
    });
  }
}
