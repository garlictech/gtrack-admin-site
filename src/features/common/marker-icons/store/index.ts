import * as markerIconsActions from './actions';
export type MarkerIconsAction = markerIconsActions.AllActions;
export { markerIconsActions };

export { State } from './state';

export * from './reducer';

export * from './selectors';
