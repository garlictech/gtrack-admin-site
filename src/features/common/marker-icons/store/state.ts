import { EntityState } from '@ngrx/entity';

export interface SvgContentEntityState extends EntityState<SvgContent> {}

export interface SvgContent {
  id: string;
  content: string;
}
export interface State {
  icons: SvgContentEntityState;
  markers: SvgContentEntityState;
}

export const featureName = 'features.common.marker-icons';
