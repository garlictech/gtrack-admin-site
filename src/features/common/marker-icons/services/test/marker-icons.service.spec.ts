import { SvgIconRegistryService } from 'angular-svg-icon';
import { of } from 'rxjs';

import { fakeAsync, TestBed } from '@angular/core/testing';
import { markerIconsReducer } from '@bit/garlictech.angular-features.common.marker-icons/store';
import { Store, StoreModule } from '@ngrx/store';

import { EIconStyle } from '../../enums';
import { featureName } from '../../store/state';
import { MarkerIconsService } from '../marker-icons.service';

describe('MarkerIconsService', () => {
  let markerIconsService: MarkerIconsService;
  let store: Store<any>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          [featureName]: markerIconsReducer
        })
      ],
      providers: [
        MarkerIconsService,
        {
          provide: SvgIconRegistryService,
          useValue: {
            loadSvg: jest.fn((path: string) => of({ outerHTML: `Mock SVG content from ${path}` }))
          }
        }
      ]
    });

    store = TestBed.get(Store);
    store.dispatch = jest.fn();
    markerIconsService = TestBed.get(MarkerIconsService);
  });

  it('should be created', () => {
    expect(markerIconsService).toBeTruthy();
  });

  it('should load SVG content to store on create', () => {
    expect((store.dispatch as any).mock.calls).toMatchSnapshot();
  });

  it('should get icon content from store', () => {
    const mockContent = 'Mock SVG content from /assets/icon/bank.svg';
    store.pipe = jest.fn().mockReturnValue(of(mockContent));

    const svg = markerIconsService.getIcon('bank', false, EIconStyle.DEFAULT);

    expect(svg).toEqual(mockContent);
  });

  it('should get marker content from store', () => {
    const mockContent = 'Mock SVG content from /assets/marker/bank.svg';
    store.pipe = jest.fn().mockReturnValue(of(mockContent));

    const svg = markerIconsService.getMarker('bank', false, EIconStyle.DEFAULT);

    expect(svg).toEqual(mockContent);
  });
});
