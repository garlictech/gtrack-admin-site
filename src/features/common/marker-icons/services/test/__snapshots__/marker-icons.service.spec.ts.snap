// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerIconsService should load SVG content to store on create 1`] = `Array []`;
